Document distributions in the 6.1 upgrade guide.

mauritsvanrees opened the pull request:
See https://github.com/plone/Products.CMFPlone/issues/3854


<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1733.org.readthedocs.build/

<!-- readthedocs-preview plone6 end -->

mauritsvanrees committed:
Document distributions in the 6.1 upgrade guide.

See https://github.com/plone/Products.CMFPlone/issues/3854

stevepiercy commented on docs/backend/upgrading/version-specific-migration/images/zope-root-distributions.png:
Is this a screenshot of the actual new distribution chooser or a compiled image?

If so, I'd like to rename this file, because "zope" is strange. `distribution-chooser.png` works for me.

Also I want to update the labels and buttons for each. Where can I make a PR to update them?

- label: Plone Site -> Plone Site (Volto)
- button: default -> Volto
- button: classic -> Classic UI

We need a new page in the documentation "Help me choose a frontend", and add a help icon (❔) that links to that page. It would highlight the differences between the two frontends for the two largest audiences. For starters:

## Volto

### Developers and Integrators

- React-based JavaScript frontend
- REST API to the Plone backend
- Python skills not required, but helpful for modifying the backend
- Accurate and current documentation in one place
- Customization of themes is well documented and relatively easier compared to Classic UI

### Editors and other end users

- Responsive, modern, fast inline editor
- Flexible layouts with blocks
- Minimal User Manual consisting only of Blocks

## Classic UI

### Developers and Integrators

- Python-based server-side templates for the frontend
- Customization of themes is not well-documented and relatively difficult compared to Volto
- Documentation may be out of date and scattered across multiple repositories

### Editors and other end users

- WYSIWYG editor for entire pages
- Rigid layouts, based on content types
- Comprehensive User Manual for Plone 5, and nothing for Plone 6

stevepiercy replied:
I moved the comments for choosing a frontend into the long neglected issue at https://github.com/plone/documentation/issues/1353#issuecomment-2401760150, so as not to block this PR.

stevepiercy replied:
@mauritsvanrees where are the source files for the UI for the frontend chooser? Sorry, I have insufficient familiarity with the usage of a distribution, and the discrepancies I mentioned should be fixed.

Then I'd need to run through the process. I'm not sure which installation method yields the screenshot you provided.

stevepiercy replied:
Resolved in #1749

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
I'll edit this file later today. Please hold on merging until I get the chance.

stevepiercy reviewed (changes requested):
Here's an initial review. I have more work to do, so please do not merge until I have a chance. Thank you!

stevepiercy committed:
Review of Distributions documentation

stevepiercy committed:
Align term definition with upgrade guide docs, avoiding circular reference.

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,80 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+
+-   The configuration form is simpler and shorter.
+-   The created site has no content, and therefore no {guilabel}`News` or {guilabel}`Events` folders.
+-   You must install then activate add-ons through the {guilabel}`Add-ons` control panel.
```
The original paragraph was unclear to me, so I attempted to clarify it with a shortened paragraph and bullet points. Please let me know if this works for y'all.

mauritsvanrees replied:
Yes, that works.
But what I tried to do was apparently not clear.  I did not want to describe the difference between CMFPlone 6.1 and a distribution, but the difference between CMFPlone 6.0 and 6.1. 

* In CMFPlone 6.0 and 6.1 the standard form is the same, except that in 6.1 the option for install the default content is gone.  That is covered by bullet two.
* In 6.0 you had the option of going to an advanced form where you could select a dozen or so extra addons/profiles for installation.  This advanced form is gone in 6.1.  This is basically covered by your last bullet point.  That bullet point does not mention the missing advanced form, but that may be fine.

I have slightly modified the text so I think it now correctly describes both differences: CMFPlone 6.1 compared to distributions and compared to CMFPlone 6.0.

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,80 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+    This package was already present in Plone 6.0, but was upgraded to a distribution.
+-   [`plone.classicui`](https://github.com/plone/plone.classicui) is a new package and is the distribution to create a Plone site with the Classic UI frontend.
+
+When you start a new project with `Plone` 6.1, the launch screen prompts you to choose a frontend distribution when creating a new Plone site.
```
We need to document the steps after `pip install Plone==6.1.0a5` (latest alpha version to try) and before this screenshot.

I have no idea how to get to this screenshot from this upgrade guide. It should go into the [Install documentation](https://6.docs.plone.org/install/), and add a link to that from here.

Also where does this fit into Cookieplone?

mauritsvanrees replied:
Start Plone (`bin/instance fg`, `docker compose up`, `make start` or whatever) and go to localhost:8080.
Calling it the launch screen helps.

I would say: if someone is reading the upgrade guide, then they already have a Plone site and know how to start it, so we don't need to describe it here.

Well, "start a new project" is probably a bad choice of words when what I really mean is just "start Plone".  Let me change that.

mauritsvanrees replied:
O, I see here and in a few other lines that you call it "frontend distributions".  That is not what they are, they are distributions installed on the backend.  It is just that in the default case your choice of distribution will be guided by what frontend you want.  If you are going to use the Volto frontend, you must install the `default` distribution (plone.volto).  If you are going to use Classic UI, you must install the `classic` distribution.

The tricky thing of course is that the `classic` distribution gives you a frontend as well, as this is the server side rendered frontend that is baked into the backend...

A Plone agency might very well love Classic UI and have a demo site where at startup you can choose between five distributions that are all variations of the `classic` distribution: website, intranet, extranet, web shop, university, etc.

I have renamed "frontend distribution" to "distribution" everywhere.

stevepiercy replied:
Well, the thing is, Classic UI is a frontend, even though it is generated on the backend, just as Node.js generates the frontend for Volto.

> Well, "start a new project" is probably a bad choice of words when what I really mean is just "start Plone". Let me change that.

Yes, that's what threw me. We're on the same page now for this item.

stevepiercy committed:
Fix admonition syntax

stevepiercy committed:
Add link to repo

stevepiercy committed:
Enhance screenshot with card syntax

stevepiercy reviewed (changes requested):
Thanks for the docs! I edited them to clarify a few things, add a few enhancements, and style.

However I still have one critical concern that I firmly believe should be addressed prior to the final release of 6.1: there are no usage docs! I cannot stress enough the importance of this, and I hope my fellow Plonistas agree.

Aside from that, it would be good for several other pairs of eyes review my revisions.

stevepiercy committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

mauritsvanrees committed:
plone.exportimport is not for upgrading Plone.

If you need an export/import for upgrading Plone you would still use collective.exportimport.
Upgrading is not a use case that plone.exportimport covers.

stevepiercy committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

mauritsvanrees committed:
Here we do not start a new project, I just mean: start Plone.

mauritsvanrees committed:
These are not 'frontend distributions'.

mauritsvanrees committed:
Creating a 6.1 CMFPlone site differs from 6.0, that is the main point here.

mauritsvanrees committed:
Don't mention installing add-ons here, activating is enough.

Sure, the Python package of an add-on must be installed before you can activate it, but the same is true for a distribution.
The plone.classicui distribution has basically the same Python package dependencies as CMFPlone, although it activates a few extra ones automatically if they are available (say plone.app.caching).

mauritsvanrees commented:
> However I still have one critical concern that I firmly believe should be addressed prior to the final release of 6.1: there are no usage docs! I cannot stress enough the importance of this, and I hope my fellow Plonistas agree.

Just checking.  What do you mean with "usage docs" in this case?  Usage docs specifically for distributions, so how to create a new distribution and how to create a site with a distribution, integrating this with the installation docs?  Or end user documentation of Plone as a whole?

stevepiercy commented:
> Just checking. What do you mean with "usage docs" in this case?

Specifically usage of `plone.distribution`, primarily for developers.

stevepiercy committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

stevepiercy commented:
BTW, the user manual is in a sad state, but at least we have a PLIP: https://github.com/plone/Products.CMFPlone/issues/3987

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+A Plone distribution is a Python package that defines specific features, themes, add-ons, and configurations that get activated when creating a Plone site.
+Now it is available in core Plone as the recommended way for creating a Plone site.
+
+When you install the [`Plone` 6.1 Python package](https://pypi.org/project/Plone/), you get several new packages and two distributions:
```
We need to reference how someone may install Plone 6.1. I would suggest something like this, since this is all we have at this moment. It's bad because it doesn't cover Classic UI.

```suggestion
When you [install the `Plone` 6.1 Python package](https://6.docs.plone.org/install/), you get several new packages and two distributions:
```

From there, we have [Create a project with Volto (development or pre-release)](https://6.docs.plone.org/install/create-project-cookieplone.html) (a very recent addition) but nothing for Classic UI for Plone 6.1. 😱 

----

As a side note, IMO, this is a critical problem that should block the release of Plone 6.1 final. Installation of Plone 6.1 and major development blockers—specifically how to write tests, create an add-on, and customize a theme—should be resolved. I don't know if anyone shares this opinion.

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
`````diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+_Launch screen for choosing a distribution_
+````
+
+After you select a distribution, you will fill out a brief form to configure your new Plone 6.1 site, similar to the process for Plone 6.0.
`````
I'm reconsidering lines 137-148 for inclusion in the Upgrade Guide. I think they belong instead in one of the [Install](https://6.docs.plone.org/install/) guides, but I'm not sure which, if any of them. I think the UI is going to change with future 6.1.x, and we almost never update an Upgrade Guide after its release, so I don't want these details and screenshot to become stale.

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+After you select a distribution, you will fill out a brief form to configure your new Plone 6.1 site, similar to the process for Plone 6.0.
+
+Distributions are optional.
+If your project only uses the `Products.CMFPlone` Python package, you can still create a Plone site in the usual way.
```
We need to clarify what is "the usual way". plone-cookie-starter, buildout, or ...?

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+
+There are a few things you should consider when upgrading a project to, or making an add-on compatible with, Plone 6.1:
+
+-   In general, you don't need no change anything.
```
```suggestion
-   In general, you don't need to change anything.
```

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+    Your existing site will keep working.
+    But adding a new site may change in the ways described earlier.
+-   Do you want to use the `Products.CMFPlone` package (0 distributions), either `plone.volto` or `plone.classicui` (1 distribution), or `Plone` (2 distributions)?
+-   When your site used the default Volto frontend, you will already have `plone.volto` as a dependency.
```
```suggestion
-   When your site uses the default Volto frontend, you already have `plone.volto` as a dependency.
```

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+-   When your site uses only the `Products.CMFPlone` package, the frontend is Classic UI.
+    This can stay the same, but you may want to depend on `plone.classicui`.
+    With that package you can still create a new site and have the same content as before.
+-   When your site is using the `Plone` package, you will have the two new distributions available.
```
```suggestion
-   When your site uses the `Plone` package, you have the two new distributions available.
```

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+    With that package you can still create a new site and have the same content as before.
+-   When your site is using the `Plone` package, you will have the two new distributions available.
+    This is fine.
+    If you know you only need `plone.volto` or only need `plone.classicui`, you can switch to that.
```
```suggestion
    If you know you only need `plone.volto`, or only need `plone.classicui`, you can switch to only the one.
```

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+    Set `ALLOWED_DISTRIBUTIONS=default` for the distribution targeting the default Volto frontend (`plone.volto`).
+    Set `ALLOWED_DISTRIBUTIONS=classic` for the distribution with the Classic UI frontend (`plone.classicui`).
+-   If you switch from `Plone` to `plone.volto` or `plone.classicui`, you might want to install extra core add-ons, for example `plone.app.upgrade` or `plone.app.caching`.
+-   If your add-on is only for Volto, you may want to add `plone.volto` as a dependency.
```
```suggestion
-   If your add-on is only for Volto, you might want to add `plone.volto` as a dependency.
```

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+    Set `ALLOWED_DISTRIBUTIONS=classic` for the distribution with the Classic UI frontend (`plone.classicui`).
+-   If you switch from `Plone` to `plone.volto` or `plone.classicui`, you might want to install extra core add-ons, for example `plone.app.upgrade` or `plone.app.caching`.
+-   If your add-on is only for Volto, you may want to add `plone.volto` as a dependency.
+-   If your add-on is only for Classic UI, you may want to add `plone.classicui` as a dependency.
```
```suggestion
-   If your add-on is only for Classic UI, you might want to add `plone.classicui` as a dependency.
```

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
````diff
@@ -117,3 +117,85 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+
+```{note}
+For Plone 7, the [Plone roadmap](https://plone.org/why-plone/roadmap) guides development toward a clearer separation between the Classic UI frontend and the core `Products.CMFPlone` backend.
+This means in Plone 7 `Products.CMFPlone` will have less code and pull in fewer dependencies.
````
```suggestion
This means in Plone 7, `Products.CMFPlone` will have less code and pull in fewer dependencies.
```

stevepiercy reviewed (changes requested):
A couple of big concerns and questions, and the rest minor grammar fixes.

MrTango committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

davisagli commented:
@mauritsvanrees @stevepiercy This is the upgrade guide, so the audience is people who are upgrading an existing site, not creating a new one.

I think it would help if we break thing apart into a few different pages:

1. A conceptual guide "Plone distributions" which explains what a distribution is (i.e. this section from the plone.distribution readme: https://github.com/plone/plone.distribution/?tab=readme-ov-file#what-is-a-plone-distribution) and the packages in Plone that are related to distributions (i.e. lines 130-135 here)
2. A how-to guide "Create a Plone site" which shows how to add a Plone site to an existing instance (regardless of how it was installed). This can be adapted from lines 137-148 here. I think it should go in the new Admin Guide which I am working on in https://github.com/plone/documentation/pull/1746
3. A how-to guide "Create a Plone distribution" which explains how to create a new Plone distribution. This can be mostly copied from the plone.distribution readme. I think it should go in the new Developer Guide.
4. This section of the upgrade guide can focus on the optional changes that can be made to an existing site (i.e. lines 158-178 here.

I can help prepare these changes if they sound good. Let me know what you think.

mauritsvanrees commented:
That sounds good to me.

stevepiercy commented:
@davisagli yes, please! I was struggling with the content straddling multiple axes of the Diataxis Framework, and I think you got it all laid it.

davisagli commented:
@stevepiercy @mauritsvanrees I didn't have time to work on it this evening, but I should be able to tackle it tomorrow.

davisagli committed:
reorganize distribution docs

davisagli commented:
@mauritsvanrees @stevepiercy @ericof  Done and ready for review.

davisagli commented on docs/developer-guide/create-a-distribution.md:
@stevepiercy The developer guide doesn't actually exist yet. This can go somewhere else temporarily but I wasn't sure where.

stevepiercy replied:
@davisagli this location is fine for now. It needs an a new file and an entry in `docs/index.md`. I'll start a review later today. I've run out of steam for tonight.

stevepiercy replied:
Done in d36830f93ff478d9ef7830c866ceaa27d50282df. We'll fiddle with it after the big Admin Guide is merged.

ericof commented on docs/developer-guide/create-a-distribution.md:
````diff
@@ -0,0 +1,294 @@
+
+```python
+        "Framework :: Plone",
+        "Framework :: Plone :: 6.0",
````
We should update this to 6.1

davisagli replied:
```suggestion
        "Framework :: Plone :: 6.1",
```

ericof commented on docs/developer-guide/create-a-distribution.md:
````diff
@@ -0,0 +1,294 @@
+
+```python
+    install_requires=[
+        "Plone",
````
This should be Products.CMFPlone

davisagli replied:
```suggestion
        "Products.CMFPlone",
```

ericof reviewed (changes requested):
Just two minor changes

davisagli committed:
Apply suggestions from code review

davisagli committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

stevepiercy committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

stevepiercy commented on docs/install/add-site.md:
```diff
@@ -0,0 +1,75 @@
+    "keywords": "Plone 6, create, add, factory, distributions"
+---
+
+% This should move into the Admin guide once https://github.com/plone/documentation/pull/1746 is merged.
```
Yup.

stevepiercy commented on docs/install/index.md:
```diff
@@ -56,4 +56,5 @@ create-project
 create-project-classic-ui
 create-project-cookieplone
 containers/index
+add-site
```
This should be moved into `admin-guide`.

stevepiercy replied:
Done in 606af43a47bc9afbfba59478db331e3d668928f8

stevepiercy commented on docs/conceptual-guides/distributions.md:
```diff
@@ -0,0 +1,79 @@
+
+Plone comes with two built-in distributions, which correspond to the two Plone user interfaces:
+- `default` (Volto)
+- `classic` (Classic UI)
```
This does not align with the Upgrade Guide at https://github.com/plone/documentation/pull/1733/files#diff-42e12720bfe48faeddf3de3f3234ec32ae6d766dfeeaaeb4952f7a4e44efb0e5R145. Where did these two names come from? We should link them to their sources.

stevepiercy replied:
Addressed in e25b72dcb9c3b1cd5fdf07f4730a2c74fba301bf, but should be double-checked for accuracy. I'm not sure where `default` and `classic` come from.

stevepiercy committed:
Update conceptual guide for distributions, where the significant changes are:
- Use the actual package names of the built-in distributions
- Move the comparison to other CMSs to the end of the page, putting Plone-only content first.

stevepiercy commented on docs/conceptual-guides/distributions.md:
```diff
@@ -0,0 +1,85 @@
+:   Create a Plone site with the Volto frontend.
+
+[`plone.classicui`](https://github.com/plone/plone.classicui) 
+:   Create a Plone site with the Classic UI frontend.
```
I replaced `default` and `classic` with these two, because they did not align with the Upgrade Guide at https://github.com/plone/documentation/pull/1733/files#diff-42e12720bfe48faeddf3de3f3234ec32ae6d766dfeeaaeb4952f7a4e44efb0e5R145.

These two should be double-checked for accuracy. I'm not sure where `default` and `classic` come from.

stevepiercy committed:
Create an entry in `index.md` to include a new `developer-guide/index.md` and the new `developer-guide/create-a-distribution.md`

stevepiercy committed:
Update glossary reference to other distribution material

stevepiercy committed:
Add cross-reference

davisagli commented on docs/conceptual-guides/distributions.md:
```diff
@@ -0,0 +1,89 @@
+
+[`plone.classicui`](https://github.com/plone/plone.classicui) 
+:   Create a Plone site with the Classic UI frontend.
+
```
@stevepiercy I don't feel super strongly about this, but I had intentionally used the names of the distributions themselves, not the names of the Python packages which are used to distribute them.

See:

- https://github.com/plone/plone.volto/blob/main/src/plone/volto/distributions.zcml#L9
- https://github.com/plone/plone.classicui/blob/main/src/plone/classicui/distributions.zcml#L7

<img width="889" alt="Screenshot 2024-11-03 at 7 43 58 PM" src="https://github.com/user-attachments/assets/b4df424c-9621-47c4-9ab9-75f4d808b438">

stevepiercy replied:
Yeah, let's keep the package names because those are not ambiguous. We programmers get lazy and shorten names, and the end users suffer for it.

I strongly dislike both "default" for "Volto" and "classic" for "Classic UI". I see no benefit to rename things. I see harm in that we have to explain what "default" and "classic" actually mean. This is especially confusing to Classic UI users because to them Classic UI _is_ the default frontend, and Volto is optional.

I intend to submit a PR to both improve its accessibility and change the text and button labels accordingly, after I make it through all the docs PRs. Those links are helpful, as I had no idea where to begin.

Here's what I intend to suggest, but without all the horizontal rules that are from the GitHub Markdown render of my comment and existing UI. I'll have to look up ZCML syntax for button accessibility attributes, too. Finally, the hover over each panel is bad for a11y. We should always show the Create buttons.

----

# Create a new Plone site

## Select an available distribution for your site's frontend

Volto [Select Volto] | Classic UI [Select Classic UI]

davisagli replied:
@stevepiercy I was already thinking `default` is a bad name for the distribution that's configured in plone.volto. "default" is a status more than an identity. And since we have multiple installation methods, there are different defaults in different situations! I'll prepare a PR in plone.volto to change the internal name of the distribution to `volto`.

The launch page is implemented as a standalone js app in https://github.com/plone/plone.distribution/tree/main/frontend ... you might need to get some tips from Érico on how to do dev on it, since I don't see anything about that in the readme for the package.

stevepiercy replied:
I created the following issues:

- https://github.com/plone/plone.distribution/issues/99
- https://github.com/plone/plone.volto/issues/161
- https://github.com/plone/plone.classicui/issues/18

Resolving this conversation.

stevepiercy committed:
Add JSON Schema to Glossary

stevepiercy committed:
Tidy up create-a-distribution.md.
- Rewrite passive voice to active.
- Enhance code blocks with `emphasize-lines`.
- Use definition lists as appropriate for terms and inline literals.
- Add links and terms where helpful.

davisagli commented on docs/developer-guide/create-a-distribution.md:
```diff
@@ -0,0 +1,323 @@
+:   Create a Plone site with the Volto frontend.
+
+[`plone.classicui`](https://github.com/plone/plone.classicui) 
+:   Create a Plone site with the Classic UI frontend.
```
@stevepiercy I do object to switching to the package names here, because it's the distribution names, not the package names, that are used in the ALLOWED_DISTRIBUTIONS variable.

See also https://github.com/plone/documentation/pull/1733/files#r1827159965

stevepiercy replied:
For the docs, I'll revert this.

However, the names for the distributions obfuscate what they contain. Can they be renamed to align with their common and familiar names without too much pain?

stevepiercy replied:
Done in 28d42eccfe40e4e95a397025dba86a30a5dcf1b8

stevepiercy commented on docs/backend/upgrading/version-specific-migration/images/distribution-chooser.png:
What do I have to do to see this new launch screen? I mean, there should be _something_ for how to use one of the install methods to see this, right? None of the methods I tried worked, including Cookieplone, Buildout, and pip.

stevepiercy committed:
Revert distribution names and enhance `ALLOWED_DISTRIBUTIONS` usage.

stevepiercy committed:
Tidy up add-site.md, mostly docs style guide stuff, and a rewording of the final admonition.
Add plural `distributions` to Glossary.

stevepiercy commented on docs/install/add-site.md:
```diff
@@ -0,0 +1,76 @@
+
+# Add a Plone site
+
+This section explains how to add a Plone site to an existing Zope instance.
```
"Zope instance" is a term that comes out of nowhere and is not defined in the glossary. Do we have a better term? Can we call it a "Plone instance"? The glossary contains the phrase "Zope WSGI instance" in two of its definitions, but is not defined itself.

davisagli replied:
I think "Zope instance" is probably a concept that's worth including in the glossary. Something along the lines of:

> Zope is an open-source application server written in Python. A Zope instance is a particular set of configuration for running Zope. A new Zope instance can be created using cookiecutter-zope-instance.

"Plone instance" could be confusing, since it's possible to add multiple Plone sites in the same Zope instance.

"Zope WSGI instance" just sounds weird to me. WSGI is one way (now the only way) to publish a Zope instance over HTTP, but it's no longer a special kind of Zope instance and it's certainly not a brand.

stevepiercy replied:
Done in 9267edbb925208f4fee7569cc06fc52913c1a732.

We use "Plone instance" and "Zope / Plone instance" in several places throughout the docs. I created a new issue to handle that: https://github.com/plone/documentation/issues/1751

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,44 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+
+Plone 6.1 introduces the concept of a Plone {term}`distribution`.
+A Plone distribution is a Python package that defines specific features, themes, add-ons, and configurations that get activated when creating a Plone site.
+Now it is available in core Plone as the recommended way for {doc}`creating a new Plone site </install/add-site>`.
```
I found no way to actually run `plone.distribution` at the suggested page, as the target states:

```
assumes that you have already {doc}`installed </install/index>` and started Plone.
```

...and none of those options pull in a 6.1 site, and neither in the currently open PRs for docs.

What did I miss?

davisagli replied:
@stevepiercy You don't "run plone.distribution" per se. It's just there if you install the `Plone` package (which is what all the documented install methods do) rather than the more minimal `Products.CMFPlone` package.

It's by design that the install methods as documented still use Plone 6.0, since Plone 6.1 is still pre-final release.

With the Cookieplone project template, be sure to choose Plone 6.1.0b1:
```
  [7/17] Should we use prerelease versions? (No): Yes
  [8/17] Plone Version (6.1.0b1): 
```

With buildout- or pip-based installs, be sure to use `https://dist.plone.org/release/6.1-latest/` instead of `https://dist.plone.org/release/6-latest/`

stevepiercy replied:
I've updated the checklist for updates to docs for Plone 6.1 final:

https://github.com/plone/documentation/issues/1721

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,44 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+
+Distributions are optional.
+If your project only uses the `Products.CMFPlone` Python package, you can still create a Plone site in the old way.
+Consider, however, the following differences from distributions and from Plone 6.0.
```
This sentence confused the heck out of me, and I only understood its intent after reading the list and already knowing some differences. I rewrote it for clarity. I hope it makes more sense now.

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-61.md:
```diff
@@ -117,3 +117,44 @@ If you have an existing Plone 5.2 or 6.0 site and you migrate to 6.1, then migra
+-   Do you want to use the `Products.CMFPlone` package (0 distributions), either `plone.volto` or `plone.classicui` (1 distribution), or `Plone` (2 distributions)?
+-   If your site uses the default Volto frontend, you will already have `plone.volto` as a dependency.
+    This can stay the same.
+-   If your site uses only the `Products.CMFPlone` package, the frontend is Classic UI.
```
I thought all Plone sites use `Products.CMFPlone`? Should this be:

```suggestion
-   If your site uses Classic UI for the frontend, you will already have `Products.CMFPlone` as a dependency and not `plone.volto`.
```

davisagli replied:
This line is talking about the situation where your project _only_ installs `Products.CMFPlone` and not `Plone`. Your suggestion as worded is not quite the point Maurits was trying to make here. Instead I suggest:
```suggestion
-   If your site depends on the `Products.CMFPlone` package without the `Plone` or `plone.volto` packages, then the frontend is Classic UI.
```

stevepiercy committed:
Tidy up distributions section in upgrade-to-61.md

stevepiercy reviewed (changes requested):
I've completed my review by pushing commits.

I made a few comments and asked some questions.

stevepiercy committed:
Update docs/backend/upgrading/version-specific-migration/upgrade-to-61.md

Co-authored-by: David Glick <david@glicksoftware.com>

stevepiercy committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

stevepiercy committed:
Move add-site into Admin Guide.
Move Admin Guide before Developer Guide in main navigation

stevepiercy committed:
Remove stray characters from bad merge

stevepiercy committed:
Resolve TODOs by moving file and adding link to new choose-user-interface.

stevepiercy committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

stevepiercy committed:
Update glossary with Zope instance and cross-references to Zope itself.

stevepiercy commented:
@davisagli I've resolved merge conflicts and made a few changes. Would you please review it again?

stevepiercy committed:
Fix cross-references

stevepiercy committed:
Fix link

stevepiercy committed:
Ignore link to static images

stevepiercy committed:
Ignore searches on GitHub as it rate limits and times out.

stevepiercy committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

davisagli committed:
Merge branch '6.0' into maurits-upgrade-61-distributions

davisagli commented on docs/developer-guide/index.md:
```diff
@@ -0,0 +1,19 @@
+
+# Developer guide
+
+This part of the documentation provides information for how to develop in Plone.
```
````suggestion
This part of the documentation provides information for how to develop in Plone.

```{note}
This part of the documentation is under construction.
You can find documentation for development in various locations through the [search feature](https://6.docs.plone.org/search.html?q=development).
You can help consolidate all of development documentation here, even if it is to let us know what is missing, by [creating an issue](https://github.com/plone/documentation/issues/new?assignees=&labels=&projects=&template=new-issue-form.yml).
```
````

davisagli reviewed (dismissed):
@stevepiercy LGTM.

stevepiercy committed:
Update docs/developer-guide/index.md

Co-authored-by: David Glick <david@glicksoftware.com>

mauritsvanrees commented:
Thanks a lot for picking this up, @davisagli and @stevepiercy !
